Direct use of Log4jLogger

jodygarnett opened the pull request:
Personal experiment follow up to https://github.com/geonetwork/core-geonetwork/pull/6397 looking at what it would take to use the Log4j Appenders API.

Core change:

* [x] Use `BasicContext.getLogger()` rather than asking `BasicContext` to implement Logger 
* [x] `LoggerWrapper` implementation (formally an anonymous class in `Log`)
* [x] Introduce [markers](https://logging.apache.org/log4j/2.x/manual/markers.html) for the modules named in `Log` (example `Log.JEEVES_MARKER`) 
* [x] Tag `AppenderWrapper` with appropriate marker for the functionality they wish to track
* [x] `Log.createLogger()` methods to support classname (or determine from stack trace)
* [x] Logger extends Log4J Logger (a lot of new efficient methods) 
* [x] review `LoggerWrapper.debug` methods to apply marker
* [x] review `LoggerWraper.info` methods to apply marker
* [x] review `LoggerWrapper.warn` methods to apply marker
* [x] review `LoggerWrapper.error` methods to apply marker 

Direct use of LOGGERS as a non critical follow up to improve over time:

* [ ] Remove references to ``Log.isDebugEnabled(module)`` (handy way to identify where loggers are needed)

The net effect should be using `Marker` to indicate geonetwork module (such as "jeeves" or "Jeeves.webapp") while having full access to the  Log4J Logger API (which is much more efficient) along with the class names which can be used in the `log4j2-dev.xml` profile.

### LOGGER Example

Although ``Log.createLogger()`` methods return ``org.fao.geonet.Logger`` it is better import ``org.apache.logging.log4j.Logger`` when possible to stick to "pure" Log4j2 API. The additional deprecated methods provided by ``org.fao.geonet.Logger`` are for things like ``warning(message)`` that can be replaced with ``warn(message)``.

```
import org.apache.logging.log4j.Logger;

...
private static Logger LOGGER = Log.createLogger(SchemaManager.class,Geonet.SCHEMA_MANAGER_MARKER);
```

### Use ParameterizedMessage format

Rather than use string concatenation or formatting using ParameterizedMessage approach:

```
LOGGER.info(Geonet.SCHEMA_MANAGER_MARKER,
  "Loading schema {} ...",
  schemaDir.getFileName() );
```

### Throwable

Pass in throwable as the last parameter and let configuration choose if a stack trace should be shown or not.
```
LOGGER.error(Geonet.SCHEMA_MANAGER_MARKER, "Cannot decode blank number from {}", baseBlank, nfe);
```

### isDebugEabled(Marker) rarely worthwhile

Because ParameratizedMethods ares so efficient:
```
LOGGER.debug(Geonet.SCHEMA_MANAGER_MARKER,
  "  => Found schema {} using AUTODETECT(attributes);examination",
  schema);
```
It replaces the need for most occupancy of ``isDebugEnabled(marker)``:
```
if (LOGGER.isDebugEnabled(Geonet.SCHEMA_MANAGER_MARKER)) {
   LOGGER.debug(Geonet.SCHEMA_MANAGER_MARKER,
     "  => Found schema " + schema + " using AUTODETECT(attributes) examination");
}
```

However if a method call is expensive you should still use isDebugEnabled(Marker):
```java
if (LOGGER.isDebugEnabled(Log.REQUEST_MARKER)) {
    LOGGER.debug(Log.REQUEST_MARKER, "Adding to parameters: {}", Xml.getString(elem));
}
```

### Use markers for readability (Optional)

Since the above ``LOGGER`` was created with ``Log.SCHEMA_MANAGER_MARKER`` the following are equivalent:

```
LOGGER.debug("No bean defined for the schema plugin '{}'.{}",
  schemaIdentifier, e.getMessage(),
  e);
```

```
LOGGER.debug(Log.SCHEMA_MANAGER_MARKER,
  "No bean defined for the schema plugin '{}'.{}", schemaIdentifier,
  e.getMessage(),
  e);
```

The second option with ``Log.SCHEMA_MANAGER_MARKER`` clearly indicates intent, and does not reply on the AppenderWrapper (allowing us opportunity to drop AppenderWrapper in the future).

jodygarnett committed:
Direct use of Log4jLogger

jodygarnett committed:
Configure log4j2.xml with marker based filters

This reproduces the previous level based filtering

jodygarnett commented:
@fxprunayre / @josegar74 I personally like this change but it is up to the team.

Right now I set the filters up at the top of the log4j2.xml file; but they could be broken up and applied to specific appenders, or loggers if that is cleaner. See https://logging.apache.org/log4j/2.x/manual/filters.html for details.

juanluisrp commented:
I don't know if it's related to this PR, but building the projects shows these warning related to `log4j-slf4j18-impl` library:
```
[INFO] Scanning for projects...
[WARNING]
[WARNING] Some problems were encountered while building the effective model for org.geonetwork-opensource:gn-camelPeriodicProducer:jar:4.2.1-SNAPSHOT
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: org.apache.logging.log4j:log4j-slf4j18-impl:jar -> version (?) vs ${log4j2.version} @ org.geonetwork-opensource:gn-workers:4.2.1-SNAPSHOT, /private/tmp/core-geonetwork/workers/pom.xml, line 105, column 17
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: org.apache.logging.log4j:log4j-slf4j18-impl:jar -> version (?) vs ${log4j2.version} @ org.geonetwork-opensource:gn-workers:4.2.1-SNAPSHOT, /private/tmp/core-geonetwork/workers/pom.xml, line 115, column 17
[WARNING]
[WARNING] Some problems were encountered while building the effective model for org.geonetwork-opensource:gn-workers:pom:4.2.1-SNAPSHOT
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: org.apache.logging.log4j:log4j-slf4j18-impl:jar -> version (?) vs ${log4j2.version} @ line 105, column 17
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: org.apache.logging.log4j:log4j-slf4j18-impl:jar -> version (?) vs ${log4j2.version} @ line 115, column 17
[WARNING]
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING]
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING]
```

juanluisrp commented:
Addressing the warning showed in the previous comment in https://github.com/geonetwork/core-geonetwork/pull/6536.

jodygarnett committed:
Migrate codebase from calling Log methods to direct use of Logger

josegar74 commented on services/src/main/java/org/fao/geonet/services/main/GenericController.java:
```diff
@@ -77,25 +78,29 @@ public void dispatch(@PathVariable String portal,
             // Log.debug(Log.REQUEST, "Char encoding: "+
             // req.getCharacterEncoding());
-            Log.debug(Log.REQUEST, "Accept       : " + request.getHeader("Accept"));
+            LOGGER.debug( "Accept       : {}",request.getHeader("Accept"));
```
Add a space after the comma

jodygarnett replied:
```suggestion
            LOGGER.debug( "Accept       : {}", request.getHeader("Accept"));
```

josegar74 commented on services/src/main/java/org/fao/geonet/services/main/GenericController.java:
```diff
@@ -77,25 +78,29 @@ public void dispatch(@PathVariable String portal,
             if (httpSession != null) {
-                Log.debug(Log.REQUEST, "Session id is " + httpSession.getId());
+
+                LOGGER.debug( "Session id is {0}",httpSession.getId());
```
I'm not sure if should be `{}` instead of `{0}`or if the last is also valid

jodygarnett replied:
It is a

josegar74 commented on web/src/main/webapp/WEB-INF/classes/setup/sql/migrate/v3110/UpdateMetadataStatus.java:
```diff
@@ -115,9 +118,14 @@ private void addMissingColumn(final Connection connection, Dialect dialect ) thr
-            Log.debug(Geonet.DB, e);
+            LOGGER.error(
+                Geonet.DB_MARKER,
+                "  Exception while adding new {} column to {}. Error is: ",
```
Seem missing a parameter `{}` for `e.getMessage()`

jodygarnett replied:
trailing attributes do not require ``{}`` but we can add for clarity
```suggestion
                "  Exception while adding new {} column to {}. Error is: {}",
```

josegar74 commented on web/src/main/webapp/WEB-INF/classes/setup/sql/migrate/v3110/UpdateMetadataStatus.java:
```diff
@@ -115,9 +118,14 @@ private void addMissingColumn(final Connection connection, Dialect dialect ) thr
+                MetadataStatus.TABLE_NAME,
+                e.getMessage()
+            );
+            LOGGER.debug(Geonet.DB_MARKER,e,e);
```
Add spaces after the commas

jodygarnett replied:
```suggestion
            LOGGER.debug(Geonet.DB_MARKER, e, e);
```

josegar74 commented on web/src/main/webapp/WEB-INF/classes/setup/sql/migrate/v3110/UpdateMetadataStatus.java:
```diff
@@ -314,9 +322,16 @@ private void finalizeChanges(final Connection connection, Dialect dialect ) thro
-            Log.debug(Geonet.DB, e, e);
+            LOGGER.error(
+                Geonet.DB_MARKER,
+                "  Exception while dropping old primary key constraint on table {}. "
```
Seem missing a parameter `{}` for `e.getMessage()`

josegar74 commented on web/src/main/webapp/WEB-INF/classes/setup/sql/migrate/v3110/UpdateMetadataStatus.java:
```diff
@@ -338,9 +360,14 @@ private void finalizeChanges(final Connection connection, Dialect dialect ) thro
+            LOGGER.error(
+                Geonet.DB_MARKER,
+                "  Exception while adding primary key on {} column for {}. "
+                    + "Error is: ",
```
Seem missing a parameter `{}` for `e.getMessage()`

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/GeonetworkDataDirectory.java:
```diff
@@ -256,10 +258,9 @@ private Path lookupProperty(JeevesServlet jeevesServlet, ServiceConfig handlerCo
-            }
+            LOGGER.debug(Geonet.DATA_DIRECTORY_MARKER,
+                " Found {} for {} with value {}",
+                    typeStr,keyToUse,value);
```
Add spaces after the commas

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -231,7 +233,7 @@ public void configure(ApplicationContext applicationContext, Path basePath, Path
                 if (!schemaDir.getFileName().toString().equals("CVS") && !schemaDir.getFileName().startsWith(".")) {
                     if (Files.isDirectory(schemaDir)) {
-                        Log.info(Geonet.SCHEMA_MANAGER, "Loading schema " + schemaDir.getFileName() + "...");
+                        LOGGER.info(Geonet.SCHEMA_MANAGER_MARKER, "Loading schema {} ...",schemaDir.getFileName());
```
Add a space after the comma

jodygarnett replied:
```suggestion
                        LOGGER.info(Geonet.SCHEMA_MANAGER_MARKER, "Loading schema {} ...", schemaDir.getFileName());
```

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -800,14 +816,15 @@ private String checkNamespace(Element md, String schema) {
-                            Log.debug(Geonet.SCHEMA_MANAGER, "  checkNamespace for dependency: " + depends.getText());
+                        LOGGER.debug(
+                            Geonet.SCHEMA_MANAGER_MARKER,
+                            "  checkNamespace for dependency: {}",depends.getText());
```
Add a space after the comma

jodygarnett replied:
```suggestion
                            "  checkNamespace for dependency: {}", depends.getText());
```

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -800,14 +816,15 @@ private String checkNamespace(Element md, String schema) {
             }
         } catch (Exception e) {
-            Log.warning(Geonet.SCHEMA_MANAGER, "Schema " + schema + " not registered?");
+            LOGGER.warn(Geonet.SCHEMA_MANAGER_MARKER, "Schema {} not registered?",schema);
```
Add a space after the comma

jodygarnett replied:
```suggestion
            LOGGER.warn(Geonet.SCHEMA_MANAGER_MARKER, "Schema {} not registered?", schema);
```

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -963,19 +980,18 @@ private void addSchema(ApplicationContext applicationContext, Path schemaDir, El
-                if (Log.isDebugEnabled(Geonet.SCHEMA_MANAGER))
-                    Log.debug(Geonet.SCHEMA_MANAGER, "Adding XmlFile " + Xml.getString(config));
+                if (LOGGER.isDebugEnabled(Geonet.SCHEMA_MANAGER_MARKER))
+                    LOGGER.debug(Geonet.SCHEMA_MANAGER_MARKER, "Adding XmlFile " + Xml.getString(config));
```
Use `{}`

jodygarnett replied:
```suggestion
                    LOGGER.debug(Geonet.SCHEMA_MANAGER_MARKER, "Adding XmlFile {}", Xml.getString(config));
```

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -963,19 +980,18 @@ private void addSchema(ApplicationContext applicationContext, Path schemaDir, El
                 config.setAttribute("file", fname);
-                if (Log.isDebugEnabled(Geonet.SCHEMA_MANAGER))
-                    Log.debug(Geonet.SCHEMA_MANAGER, "Adding XmlFile " + Xml.getString(config));
+                if (LOGGER.isDebugEnabled(Geonet.SCHEMA_MANAGER_MARKER))
```
Can be removed `LOGGER.isDebugEnabled(Geonet.SCHEMA_MANAGER_MARKER)`?

jodygarnett replied:
Nope, statement avoids call to xml string

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -986,7 +1002,7 @@ private void addSchema(ApplicationContext applicationContext, Path schemaDir, El
         mds.setOperationFilters(extractOperationFilters(xmlIdFile));
 
-        Log.debug(Geonet.SCHEMA_MANAGER, "  UUID is read/write mode: " + mds.isReadwriteUUID());
+        LOGGER.debug(Geonet.SCHEMA_MANAGER_MARKER, "  UUID is read/write mode: " + mds.isReadwriteUUID());
```
Use `{}`

jodygarnett replied:
```suggestion
        LOGGER.debug(Geonet.SCHEMA_MANAGER_MARKER, "  UUID is read/write mode: {}", mds.isReadwriteUUID());
```

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -1002,9 +1018,10 @@ private void addSchema(ApplicationContext applicationContext, Path schemaDir, El
+        LOGGER.debug(
+                Geonet.SCHEMA_MANAGER_MARKER,
+                "Property {} is {}",
+                Constants.XML_CATALOG_FILES,System.getProperty(Constants.XML_CATALOG_FILES));
```
Add space after the comma

jodygarnett replied:
```suggestion
                Constants.XML_CATALOG_FILES, System.getProperty(Constants.XML_CATALOG_FILES));
```

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -1078,8 +1095,7 @@ private Element deleteSchemaFromPluginCatalog(String name, Element root) throws
             if (!uri.getName().equals("uri") || !uri.getNamespace().equals(Namespaces.OASIS_CATALOG)) {
-                if (Log.isDebugEnabled(Geonet.SCHEMA_MANAGER))
-                    Log.debug(Geonet.SCHEMA_MANAGER, "Skipping element " + uri.getQualifiedName() + ":" + uri.getNamespace());
+                LOGGER.debug(Geonet.SCHEMA_MANAGER_MARKER, "Skipping element {}:{}",uri.getQualifiedName(), uri.getNamespace());
```
Add space after the comma

jodygarnett replied:
```suggestion
                LOGGER.debug(Geonet.SCHEMA_MANAGER_MARKER, "Skipping element {}:{}", uri.getQualifiedName(), uri.getNamespace());
```

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -1113,8 +1129,7 @@ private int getHighestSchemaPluginCatalogId(String name, Element root) throws Ex
             if (!uri.getName().equals("rewriteURI") || !uri.getNamespace().equals(Namespaces.OASIS_CATALOG)) {
-                if (Log.isDebugEnabled(Geonet.SCHEMA_MANAGER))
-                    Log.debug(Geonet.SCHEMA_MANAGER, "Skipping element " + uri.getQualifiedName() + ":" + uri.getNamespace());
+                LOGGER.debug(Geonet.SCHEMA_MANAGER_MARKER, "Skipping element {}:{}", uri.getQualifiedName(),uri.getNamespace());
```
Add space after the comma

jodygarnett replied:
```suggestion
                LOGGER.debug(Geonet.SCHEMA_MANAGER_MARKER, "Skipping element {}:{}", uri.getQualifiedName(), uri.getNamespace());
```

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -1282,12 +1293,12 @@ private void processSchema(ApplicationContext applicationContext, Path schemasDi
 
         if (!Files.exists(idFile)) {
-            Log.error(Geonet.SCHEMA_MANAGER, "    Skipping : " + schemasDir.getFileName() + " as it doesn't have " +
+            LOGGER.error(Geonet.SCHEMA_MANAGER_MARKER, "    Skipping : " + schemasDir.getFileName() + " as it doesn't have " +
```
Use `{}`

jodygarnett replied:
```suggestion
            LOGGER.error(Geonet.SCHEMA_MANAGER_MARKER, "    Skipping : {}  as it doesn't have [] ", schemasDir.getFileName() , 
```

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -1282,12 +1293,12 @@ private void processSchema(ApplicationContext applicationContext, Path schemasDi
         }
 
-        Log.info(Geonet.SCHEMA_MANAGER, "    Adding xml schema : " + schemasDir.getFileName());
+        LOGGER.info(Geonet.SCHEMA_MANAGER_MARKER, "    Adding xml schema : {}",schemasDir.getFileName());
```
Add space after the comma

jodygarnett replied:
```suggestion
        LOGGER.info(Geonet.SCHEMA_MANAGER_MARKER, "    Adding xml schema : {}", schemasDir.getFileName());
```

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -1356,7 +1367,7 @@ private void checkAppSupported(Element schemaPluginCatRoot) throws Exception {
 
             if (appVersion.compareTo(schemaMinorAppVersion) < 0) {
-                Log.error(Geonet.SCHEMA_MANAGER, "Schema " + schemaName +
+                LOGGER.error(Geonet.SCHEMA_MANAGER_MARKER, "Schema " + schemaName +
```
Use `{}`

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -1368,7 +1379,7 @@ private void checkAppSupported(Element schemaPluginCatRoot) throws Exception {
 
                 if (appVersion.compareTo(schemaMajorAppVersion) > 0) {
-                    Log.error(Geonet.SCHEMA_MANAGER, "Schema " + schemaName +
+                    LOGGER.error(Geonet.SCHEMA_MANAGER_MARKER, "Schema " + schemaName +
```
Use `{}`

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -1609,18 +1619,19 @@ private String compareElementsAndAttributes(Element md, int mode) throws SchemaM
-                if (Log.isDebugEnabled(Geonet.SCHEMA_MANAGER))
-                    Log.debug(Geonet.SCHEMA_MANAGER, "		Checking autodetect element " + Xml.getString(elem) + " with name " + elem.getName());
+                if (LOGGER.isDebugEnabled(Geonet.SCHEMA_MANAGER_MARKER))
+                    LOGGER.debug(Geonet.SCHEMA_MANAGER_MARKER, "		Checking autodetect element " + Xml.getString(elem) + " with name " + elem.getName());
```
Use `{}`

jodygarnett replied:
```suggestion
                    LOGGER.debug(Geonet.SCHEMA_MANAGER_MARKER,
                        "		Checking autodetect element {} with name {}",
                        Xml.getString(elem), elem.getName());
```

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -1609,18 +1619,19 @@ private String compareElementsAndAttributes(Element md, int mode) throws SchemaM
             for (Element elem : adElems) {
-                if (Log.isDebugEnabled(Geonet.SCHEMA_MANAGER))
-                    Log.debug(Geonet.SCHEMA_MANAGER, "		Checking autodetect element " + Xml.getString(elem) + " with name " + elem.getName());
+                if (LOGGER.isDebugEnabled(Geonet.SCHEMA_MANAGER_MARKER))
```
Can be removed `LOGGER.isDebugEnabled(Geonet.SCHEMA_MANAGER_MARKER)`?

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -1664,8 +1672,10 @@ private String compareElementsAndAttributes(Element md, int mode) throws SchemaM
                         if (mode == MODE_ROOT && type != null && "root".equals(type.getValue())) {
-                            if (Log.isDebugEnabled(Geonet.SCHEMA_MANAGER))
-                                Log.debug(Geonet.SCHEMA_MANAGER, "				Comparing " + Xml.getString(kid) + " with " + md.getName() + " with namespace " + md.getNamespace() + " : " + (kid.getName().equals(md.getName()) && kid.getNamespace().equals(md.getNamespace())));
+                            if (LOGGER.isDebugEnabled(Geonet.SCHEMA_MANAGER_MARKER))
```
Can be removed `LOGGER.isDebugEnabled(Geonet.SCHEMA_MANAGER_MARKER)`?

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -1664,8 +1672,10 @@ private String compareElementsAndAttributes(Element md, int mode) throws SchemaM
-                            if (Log.isDebugEnabled(Geonet.SCHEMA_MANAGER))
-                                Log.debug(Geonet.SCHEMA_MANAGER, "				Comparing " + Xml.getString(kid) + " with " + md.getName() + " with namespace " + md.getNamespace() + " : " + (kid.getName().equals(md.getName()) && kid.getNamespace().equals(md.getNamespace())));
+                            if (LOGGER.isDebugEnabled(Geonet.SCHEMA_MANAGER_MARKER))
+                                LOGGER.debug(
+                                    Geonet.SCHEMA_MANAGER_MARKER,
+                                    "				Comparing " + Xml.getString(kid) + " with " + md.getName() + " with namespace " + md.getNamespace() + " : " + (kid.getName().equals(md.getName()) && kid.getNamespace().equals(md.getNamespace())));
```
Use `{}`

jodygarnett replied:
```suggestion
                                LOGGER.debug(
                                    Geonet.SCHEMA_MANAGER_MARKER,
                                    "				Comparing {} with {}  with namespace {}: {}",
                                    Xml.getString(kid), md.getName(), md.getNamespace(),
                                    (kid.getName().equals(md.getName()) && kid.getNamespace().equals(md.getNamespace()))
                                );
```

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -1675,9 +1685,15 @@ private String compareElementsAndAttributes(Element md, int mode) throws SchemaM
-                            if (Log.isDebugEnabled(Geonet.SCHEMA_MANAGER))
-                                Log.debug(Geonet.SCHEMA_MANAGER, "				Comparing " + Xml.getString(kid) + " with " + md.getName() + " with namespace " + md.getNamespace() + " : " + (kid.getName().equals(md.getName()) && kid.getNamespace().equals(md.getNamespace())));
-
+                            if (LOGGER.isDebugEnabled(Geonet.SCHEMA_MANAGER_MARKER )) {
```
Can be removed `LOGGER.isDebugEnabled(Geonet.SCHEMA_MANAGER_MARKER )`?

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -1742,8 +1757,7 @@ private boolean isMatchingAttributeInMetadata(Attribute needle, Element haystack
     private boolean isMatchingNamespaceInMetadata(Namespace needle, Element haystack) {
-        if (Log.isDebugEnabled(Geonet.SCHEMA_MANAGER))
-            Log.debug(Geonet.SCHEMA_MANAGER, "Matching " + needle.toString());
+        LOGGER.debug(Geonet.SCHEMA_MANAGER_MARKER, "Matching {}", needle);
```
Is not required to call `needle.toString()`?

jodygarnett replied:
No it is not, the default is to call toString()

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -1789,23 +1803,25 @@ private boolean isMatchingElementInMetadata(Element needle, Element haystack, bo
         Namespace needleNS = needle.getNamespace();
-        if (Log.isDebugEnabled(Geonet.SCHEMA_MANAGER))
-            Log.debug(Geonet.SCHEMA_MANAGER, "Matching " + Xml.getString(needle));
+        if (LOGGER.isDebugEnabled(Geonet.SCHEMA_MANAGER_MARKER)) {
```
Can be removed `LOGGER.isDebugEnabled(Geonet.SCHEMA_MANAGER_MARKER)`?

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -1789,23 +1803,25 @@ private boolean isMatchingElementInMetadata(Element needle, Element haystack, bo
-                    }
+                    LOGGER.debug(
+                        Geonet.SCHEMA_MANAGER_MARKER,
+                        "    Pattern {} applied to value{} match: {}",
```
Add a space to separate `value{}`

jodygarnett replied:
```suggestion
                        "    Pattern {} applied to value {} match: {}",
```

josegar74 commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -1882,7 +1898,7 @@ public boolean accept(Path entry) throws IOException {
 
             if (missingXsdFiles) {
-                Log.error(Geonet.SCHEMA_MANAGER, "Schema " + name + " does not have any XSD files!");
+                LOGGER.error(Geonet.SCHEMA_MANAGER_MARKER, "Schema " + name + " does not have any XSD files!");
```
Use `{}`

jodygarnett replied:
```suggestion
                LOGGER.error(Geonet.SCHEMA_MANAGER_MARKER, "Schema {} does not have any XSD files!", name);
```

jodygarnett commented on common/src/main/java/org/fao/geonet/utils/BinaryFile.java:
```diff
@@ -114,10 +117,10 @@ private static int checkAck(InputStream in) throws IOException {
             }
             if (b == 2) { // fatal error
-                Log.error(Log.RESOURCES, "scp: Protocol error: " + sb.toString());
+                LOGGER.error("scp: Protocol error: {}",sb);
```
```suggestion
                LOGGER.error("scp: Protocol error: {}", sb);
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -129,15 +132,15 @@ public static Path registerXmlCatalogFiles(Path webappDir, Path schemapluginUriC
         catalogProp = webInf.resolve("oasis-catalog.xml") + ";" + schemapluginUriCatalog;
         System.setProperty(Constants.XML_CATALOG_FILES, catalogProp);
-        Log.info(Geonet.SCHEMA_MANAGER, Constants.XML_CATALOG_FILES + " property set to " + catalogProp);
+        LOGGER.info(Geonet.SCHEMA_MANAGER_MARKER, Constants.XML_CATALOG_FILES + " property set to " + catalogProp);
```
```suggestion
        LOGGER.info(Geonet.SCHEMA_MANAGER_MARKER, "{} property set to {}", Constants.XML_CATALOG_FILES, catalogProp);
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -129,15 +132,15 @@ public static Path registerXmlCatalogFiles(Path webappDir, Path schemapluginUriC
         Path blankXSLFile = webappDir.resolve("xsl").resolve("blanks.xsl");
         System.setProperty(Constants.XML_CATALOG_BLANKXSLFILE, blankXSLFile.toUri().toASCIIString());
-        Log.info(Geonet.SCHEMA_MANAGER, Constants.XML_CATALOG_BLANKXSLFILE + " property set to " + blankXSLFile);
+        LOGGER.info(Geonet.SCHEMA_MANAGER_MARKER, Constants.XML_CATALOG_BLANKXSLFILE + " property set to " + blankXSLFile);
```
```suggestion
        LOGGER.info(Geonet.SCHEMA_MANAGER_MARKER,  "{} property set to {}", Constants.XML_CATALOG_BLANKXSLFILE, blankXSLFile);
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -164,10 +167,9 @@ public static SchemaPlugin getSchemaPlugin(String schemaIdentifier) {
-                Log.debug(Geonet.SCHEMA_MANAGER, "No bean defined for the schema plugin '" +
-                    schemaIdentifier + "'. " +
-                    e.getMessage());
+            if (LOGGER.isDebugEnabled(Geonet.SCHEMA_MANAGER_MARKER)) {
+                LOGGER.debug(Geonet.SCHEMA_MANAGER_MARKER, "No bean defined for the schema plugin '{}'. {}",
+                    schemaIdentifier, e.getMessage(),e);
             }
```
```suggestion
            LOGGER.debug(Geonet.SCHEMA_MANAGER_MARKER,
                    "No bean defined for the schema plugin '{}'. {}",
                    schemaIdentifier, e.getMessage(), e);
```

jodygarnett reviewed (commented):
Suggesting changes from feedback

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/SvnManager.java:
```diff
@@ -144,7 +147,7 @@ public void init() throws Exception {
 
             } else {
-                Log.error(Geonet.SVN_MANAGER, "Problem creating or using repository at path " + subversionPath + ", " + e.getMessage(),  e);
+                LOGGER.error(Geonet.SVN_MANAGER_MARKER, "Problem creating or using repository at path " + subversionPath + ", " + e.getMessage(),  e);
```
```suggestion
                LOGGER.error(Geonet.SVN_MANAGER_MARKER,
                    "Problem creating or using repository at path {}, {} ",
                    subversionPath, e.getMessage(),  e);
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/SvnManager.java:
```diff
@@ -154,7 +157,7 @@ public void init() throws Exception {
             repo = getRepository();
         } catch (SVNException se) {
-            Log.error(Geonet.SVN_MANAGER, "Failed to open subversion repo at path " + subversionPath, se);
+            LOGGER.error(Geonet.SVN_MANAGER_MARKER, "Failed to open subversion repo at path " + subversionPath, se);
```
```suggestion
            LOGGER.error(Geonet.SVN_MANAGER_MARKER, "Failed to open subversion repo at path {}", subversionPath, se);
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/SvnManager.java:
```diff
@@ -173,8 +176,7 @@ public void init() throws Exception {
                 if (!uuid.equals(repoUuid)) {
-                    Log.warning(Geonet.SVN_MANAGER, "Recreating subversion repository at " + subversionPath
-                        + " as previous repository was empty");
+                    LOGGER.warn(Geonet.SVN_MANAGER_MARKER, "Recreating subversion repository at {} as previous repository was empty",subversionPath);
```
```suggestion
                    LOGGER.warn(Geonet.SVN_MANAGER_MARKER, "Recreating subversion repository at {} as previous repository was empty", subversionPath);
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/SvnManager.java:
```diff
@@ -345,20 +346,19 @@ public void setHistory(final String id, final ServiceContext context) throws Exc
+            LOGGER.debug(
+                Geonet.SVN_MANAGER_MARKER,
+                "Changes to metadata {} will be committed/aborted with the database channel {}",
+                id,status);
```
```suggestion
                id, status);
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/SvnManager.java:
```diff
@@ -345,20 +346,19 @@ public void setHistory(final String id, final ServiceContext context) throws Exc
+                id,status);
         } catch (Exception e) {
-            Log.error(Geonet.SVN_MANAGER, "Failed to set history for metadata to subversion repo at path " + subversionPath, e);
+            LOGGER.error(Geonet.SVN_MANAGER_MARKER, "Failed to set history for metadata to subversion repo at path " + subversionPath, e);
```
```suggestion
            LOGGER.error(Geonet.SVN_MANAGER_MARKER, "Failed to set history for metadata to subversion repo at path {}", subversionPath, e);
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/SvnManager.java:
```diff
@@ -402,21 +402,19 @@ public void createMetadataDir(final String id, ServiceContext context, Element m
             editor.closeEdit();
         } catch (SVNException svne) {
-            Log.error(Geonet.SVN_MANAGER, "Create metadata dir. error: " + svne.getMessage(), svne);
+            LOGGER.error(Geonet.SVN_MANAGER_MARKER, "Create metadata dir. error: " + svne.getMessage(), svne);
```
```suggestion
            LOGGER.error(Geonet.SVN_MANAGER_MARKER, "Create metadata dir. error: {}", svne.getMessage(), svne);
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/SvnManager.java:
```diff
@@ -430,14 +428,12 @@ public void createMetadataDir(final String id, ServiceContext context, Element m
 
-            if (Log.isDebugEnabled(Geonet.SVN_MANAGER))
-                Log.debug(Geonet.SVN_MANAGER, "Metadata " + id + " was added");
+            LOGGER.debug(Geonet.SVN_MANAGER_MARKER, "Metadata {} was added",id);
```
```suggestion
            LOGGER.debug(Geonet.SVN_MANAGER_MARKER, "Metadata {} was added", id);
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/SvnManager.java:
```diff
@@ -430,14 +428,12 @@ public void createMetadataDir(final String id, ServiceContext context, Element m
+            LOGGER.debug(Geonet.SVN_MANAGER_MARKER, "Commit returned {}", commitInfo);
         } catch (SVNException svne) {
-            Log.error(Geonet.SVN_MANAGER, "Create metadata dir. error: " + svne.getMessage(), svne);
+            LOGGER.error(Geonet.SVN_MANAGER_MARKER, "Create metadata dir. error: " + svne.getMessage(), svne);
```
```suggestion
            LOGGER.error(Geonet.SVN_MANAGER_MARKER, "Create metadata dir. error: {}", svne.getMessage(), svne);
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/SvnManager.java:
```diff
@@ -459,18 +455,16 @@ public void deleteDir(String id, ServiceContext context) throws Exception {
+            LOGGER.debug(Geonet.SVN_MANAGER_MARKER, "Directory for metadata {} deleted: {}", id, commitInfo);
         } catch (SVNException svne) {
-            Log.error(Geonet.SVN_MANAGER, "Delete metadata dir. error: " +  svne.getMessage(),  svne);
+            LOGGER.error(Geonet.SVN_MANAGER_MARKER, "Delete metadata dir. error: " +  svne.getMessage(),  svne);
```
```suggestion
            LOGGER.error(Geonet.SVN_MANAGER_MARKER, "Delete metadata dir. error: {}", svne.getMessage(),  svne);
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/SvnManager.java:
```diff
@@ -571,7 +565,7 @@ public void commitMetadata(String id, ISVNEditor editor) throws Exception {
             commitMetadataStatus(editor, id, dataMan);
         } catch (Exception e) {
-            Log.error(Geonet.SVN_MANAGER, "Failed to commit metadata to  subversion repo at path " + subversionPath, e);
+            LOGGER.error(Geonet.SVN_MANAGER_MARKER, "Failed to commit metadata to  subversion repo at path " + subversionPath, e);
```
```suggestion
            LOGGER.error(Geonet.SVN_MANAGER_MARKER, "Failed to commit metadata to  subversion repo at path {}", subversionPath, e);
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/SvnManager.java:
```diff
@@ -716,15 +704,12 @@ private void commitMetadataOwner(ISVNEditor editor, String id) throws Exception
                 SvnUtils.modifyFile(editor, id + "/owner.xml", old.getBytes(Constants.ENCODING), now.getBytes(Constants.ENCODING));
-                if (Log.isDebugEnabled(Geonet.SVN_MANAGER))
-                    Log.debug(Geonet.SVN_MANAGER, "Ownership of metadata " + id + " was updated");
+                LOGGER.debug(Geonet.SVN_MANAGER_MARKER, "Ownership of metadata {} was updated",id);
```
```suggestion
                LOGGER.debug(Geonet.SVN_MANAGER_MARKER, "Ownership of metadata {} was updated", id);
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/Thesaurus.java:
```diff
@@ -224,10 +226,7 @@ public FileTime getLastModifiedTime() {
-        if (Log.isDebugEnabled(Geonet.THESAURUS)) {
-            Log.debug(Geonet.THESAURUS, title + " has lastModified of: " + lastModified.toMillis());
-        }
+        LOGGER.debug(Geonet.THESAURUS_MARKER, "{} has lastModified of: {}",title, lastModified.toMillis());
```
```suggestion
        LOGGER.debug(Geonet.THESAURUS_MARKER, "{} has lastModified of: {}", title, lastModified.toMillis());
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/Thesaurus.java:
```diff
@@ -288,7 +287,7 @@ public synchronized Thesaurus initRepository() throws ConfigurationException, IO
         QueryEvaluationException, AccessDeniedException {
         if (Log.isDebugEnabled(Geonet.THESAURUS))
-            Log.debug(Geonet.THESAURUS, "Query : " + query);
+            LOGGER.debug(Geonet.THESAURUS_MARKER, "Query : " + query);
```
```suggestion
            LOGGER.debug(Geonet.THESAURUS_MARKER, "Query : {}", query);
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/oaipmh/ResumptionTokenCache.java:
```diff
@@ -51,8 +53,9 @@ public class ResumptionTokenCache extends Thread {
         this.settingMan = sm;
-        if (Log.isDebugEnabled(Geonet.OAI_HARVESTER))
-            Log.debug(Geonet.OAI_HARVESTER, "OAI cache ::init timout:" + getTimeout());
+        if (LOGGER.isDebugEnabled(Geonet.OAI_HARVESTER_MARKER)) {
```
Better not, we do not know how expensive the method call

jodygarnett reviewed (commented):
Additional suggested changes for batch; this is very labour intensive - next time this kind of activity should be on a branch for shared editing :)

jodygarnett commented on core/src/main/java/org/fao/geonet/resources/Resources.java:
```diff
@@ -298,9 +298,9 @@ public String copyLogo(ServiceContext context, String icon,
-            context.warning("Cannot copy icon -> " + e.getMessage());
-            context.warning(" (C) Source : " + icon);
-            context.warning(" (C) Destin : " + logosDir);
+            context.getLogger().warning("Cannot copy icon -> " + e.getMessage());
+            context.getLogger().warning(" (C) Source : " + icon);
+            context.getLogger().warning(" (C) Destin : " + logosDir);
```
```suggestion
            org.apache.logging.log4j.Logger LOGGER = context.getLogger();
            
            LOGGER.warning("Cannot copy icon -> {}", e.getMessage());
            LOGGER.warning(" (C) Source : {}", icon);
            LOGGER.warning(" (C) Destin : {}", logosDir);
```

jodygarnett commented on schemas/iso19139/src/main/java/org/fao/geonet/schema/iso19139/ISO19139SchemaPlugin.java:
```diff
@@ -271,10 +274,13 @@ public List<Element> getTranslationForElement(Element element, String languageId
-                " using XPath '" + path +
-                "updatedLocalizedTextElement exception " + e.getMessage());
+            LOGGER.debug(LOGGER_NAME,
+                "{}: getTranslationForElement failed on element {} using XPath '{}' updatedLocalizedTextElement exception ",
```
```suggestion
                "{}: getTranslationForElement failed on element {} using XPath '{}' updatedLocalizedTextElement exception {}",
```

jodygarnett commented on services/src/main/java/org/fao/geonet/api/AllRequestsInterceptor.java:
```diff
@@ -88,18 +90,19 @@ private void createSessionForAllButNotCrawlers(HttpServletRequest request) {
-                if (Log.isDebugEnabled(Log.REQUEST)) {
-                    Log.debug(Log.REQUEST, "Session created for client : " + request.getRemoteAddr());
+                if (LOGGER.isDebugEnabled(Log.REQUEST_MARKER)) {
+                    LOGGER.debug(Log.REQUEST_MARKER, "Session created for client : {}",request.getRemoteAddr());
```
```suggestion
                    LOGGER.debug(Log.REQUEST_MARKER, "Session created for client : {}", request.getRemoteAddr());
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -129,15 +132,15 @@ public static Path registerXmlCatalogFiles(Path webappDir, Path schemapluginUriC
         }
         if (!catalogProp.equals("")) {
-            Log.info(Geonet.SCHEMA_MANAGER, "Overriding " + Constants.XML_CATALOG_FILES + " property (was set to " + catalogProp + ")");
+            LOGGER.info(Geonet.SCHEMA_MANAGER_MARKER, "Overriding " + Constants.XML_CATALOG_FILES + " property (was set to " + catalogProp + ")");
```
```suggestion
            LOGGER.info(Geonet.SCHEMA_MANAGER_MARKER, "Overriding {}  property (was set to {})", Constants.XML_CATALOG_FILES, catalogProp);
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -1356,7 +1367,7 @@ private void checkAppSupported(Element schemaPluginCatRoot) throws Exception {
 
             if (appVersion.compareTo(schemaMinorAppVersion) < 0) {
-                Log.error(Geonet.SCHEMA_MANAGER, "Schema " + schemaName +
+                LOGGER.error(Geonet.SCHEMA_MANAGER_MARKER, "Schema " + schemaName +
                     " requires min Geonetwork version: " + minorAppVersionSupported + ", current is: " +
                     version + ". Skip load schema.");
```
```suggestion
                LOGGER.error(Geonet.SCHEMA_MANAGER_MARKER,
                    "Schema {} requires min Geonetwork version: {}, current is: {}. Skip load schema.", schemaName, minorAppVersionSupported, version);
```

jodygarnett commented on core/src/main/java/org/fao/geonet/kernel/SchemaManager.java:
```diff
@@ -1368,7 +1379,7 @@ private void checkAppSupported(Element schemaPluginCatRoot) throws Exception {
 
                 if (appVersion.compareTo(schemaMajorAppVersion) > 0) {
-                    Log.error(Geonet.SCHEMA_MANAGER, "Schema " + schemaName +
+                    LOGGER.error(Geonet.SCHEMA_MANAGER_MARKER, "Schema " + schemaName +
                         " requires max Geonetwork version: " + majorAppVersionSupported + ", current is: " +
                         version + ". Skip load schema.");
```
```suggestion
                    LOGGER.error(Geonet.SCHEMA_MANAGER_MARKER,
                        "Schema {} requires max Geonetwork version: {}, current is: {}. Skip load schema.",
                        schemaName, majorAppVersionSupported, version);
```

jodygarnett commented on services/src/main/java/org/fao/geonet/api/AllRequestsInterceptor.java:
```diff
@@ -88,18 +90,19 @@ private void createSessionForAllButNotCrawlers(HttpServletRequest request) {
-                if (Log.isDebugEnabled(Log.REQUEST)) {
-                    Log.debug(Log.REQUEST, "Session created for client : " + request.getRemoteAddr());
+                if (LOGGER.isDebugEnabled(Log.REQUEST_MARKER)) {
+                    LOGGER.debug(Log.REQUEST_MARKER, "Session created for client : {}",request.getRemoteAddr());
```
I prefer not if it is method call; may be an expensive wrapped request or something

jodygarnett commented on services/src/main/java/org/fao/geonet/api/AllRequestsInterceptor.java:
```diff
@@ -88,18 +90,19 @@ private void createSessionForAllButNotCrawlers(HttpServletRequest request) {
 
-                if (Log.isDebugEnabled(Log.REQUEST)) {
-                    Log.debug(Log.REQUEST, "Session created for client : " + request.getRemoteAddr());
+                if (LOGGER.isDebugEnabled(Log.REQUEST_MARKER)) {
```
I prefer not if it is method call; may be an expensive wrapped request or something

jodygarnett commented on services/src/main/java/org/fao/geonet/api/mapservers/GeoServerRest.java:
```diff
@@ -505,19 +524,23 @@ public boolean createStyle(String ws, String layer, String sldbody) {
-                    + layer + " in workspace " + ws + ", error is: " + e.getMessage());
+            LOGGER.debug(
+                LOGGER_MARKER,
+                "Failed to create style for layer: {} in workspace {}, error is: ",
```
```suggestion
                "Failed to create style for layer: {} in workspace {}, error is: {}",
```

jodygarnett commented on services/src/main/java/org/fao/geonet/api/mapservers/GeoServerRest.java:
```diff
@@ -561,10 +584,15 @@ public boolean createFeatureType(String ws, String ds, String ft,
+        LOGGER.debug(
+            Geonet.GEOPUBLISH_MARKER,
+            "Checking if a featuretype named {} already exists in workspace {}, datastore {}",
+            ft,ws,ds
```
```suggestion
            ft, ws, ds
```

jodygarnett commented on services/src/main/java/org/fao/geonet/api/mapservers/GeoServerRest.java:
```diff
@@ -561,10 +584,15 @@ public boolean createFeatureType(String ws, String ds, String ft,
         if (status != 200) {
-            Log.debug(Geonet.GEOPUBLISH, "Creating featuretype " + ft + " in workspace " + ws + " within datastore " + ds);
+            LOGGER.debug(Geonet.GEOPUBLISH_MARKER, "Creating featuretype {} in workspace {} within datastore {}",
+                ft,ws,ds);
```
```suggestion
                ft, ws, ds);
```

jodygarnett commented on services/src/main/java/org/fao/geonet/api/mapservers/GeoServerRest.java:
```diff
@@ -667,16 +695,14 @@ int sendREST(String method, String urlParams, String postData,
             m.addHeader(new BasicScheme().authenticate(new UsernamePasswordCredentials(username, password), m));
         } catch (AuthenticationException a) {
-            Log.warning(LOGGER_NAME, "Failed to add the authentication Header, error is: " + a.getMessage());
+            LOGGER.warn(LOGGER_MARKER, "Failed to add the authentication Header, error is: {}",a.getMessage());
```
```suggestion
            LOGGER.warn(LOGGER_MARKER, "Failed to add the authentication Header, error is: {}", a.getMessage());
```

jodygarnett commented on csw-server/src/main/java/org/fao/geonet/component/csw/DescribeRecord.java:
```diff
@@ -224,7 +224,7 @@ private HashMap<String, Element> getSchemaComponents(ServiceContext context, Str
                     scElements.put(tname, currentSC);
                 } catch (Exception ex) {
-                    context.warning("Error while getting schema " + tname + " (" + schema+"): " + ex.getMessage());
+                    context.getLogger().warning("Error while getting schema " + tname + " (" + schema+"): " + ex.getMessage());
```
```suggestion
                    context.getLogger().warning("Error while getting schema {} ({}): {}", tname, schema, ex.getMessage());
```

jodygarnett replied:
```suggestion
                    context.getLogger().warning("Error while getting schema {} ({}): {}", tname, schema, ex.getMessage());
```

jodygarnett commented on common/src/main/java/org/fao/geonet/utils/BinaryFile.java:
```diff
@@ -114,10 +117,10 @@ private static int checkAck(InputStream in) throws IOException {
             }
             if (b == 2) { // fatal error
-                Log.error(Log.RESOURCES, "scp: Protocol error: " + sb.toString());
+                LOGGER.error("scp: Protocol error: {}",sb);
```
```suggestion
                LOGGER.error("scp: Protocol error: {}", sb);
```

jodygarnett replied:
```suggestion
                LOGGER.error("scp: Protocol error: {}", sb);
```

jodygarnett reviewed (commented):
The GitHub PR conversations are struggling with so many small changes

jodygarnett commented on common/src/main/java/org/fao/geonet/utils/BinaryFile.java:
```diff
@@ -114,10 +117,10 @@ private static int checkAck(InputStream in) throws IOException {
             while (c != '\n');
             if (b == 1) { // error
-                Log.error(Log.RESOURCES, "scp: Protocol error: " + sb.toString());
+                LOGGER.error("scp: Protocol error: {}",sb);
```
```suggestion
                LOGGER.error("scp: Protocol error: {}", sb);
```

jodygarnett committed:
Revise based on feedback

- Most of these changes are code formatting only.
- When doing a pattern message you can use {} or {1}
- When doing pattern message any additional objects are appended (so trailing {} is optional)
- There are still log messages made by appending strings, rather than using a pattern message with fill-in-the-{}-placeholders. They can be addressed over time
- Testing with LOGGER.isDebugEnabled(marker) is not required so much anymore, however if your log message contains a possibly expensive method it is still a good idea

jodygarnett commented on common/src/main/java/org/fao/geonet/utils/BinaryFile.java:
```diff
@@ -271,7 +273,7 @@ private String readInput(Path path) {
             return new String(Files.readAllBytes(path), Constants.CHARSET);
         } catch (IOException e) {
-            Log.error("geonetwork", "Error reading file: " + path);
+            LOGGER.error(MarkerManager.getMarker("geonetwork"), "Error reading file: {}", path);
```
added a comment indicating why a string is being used

jodygarnett committed:
Continue to replace additional Log.isDebugEnabled calls

jodygarnett commented:
rebased; this should be good to go

jodygarnett commented:
@josegar74 identified TRACE methods coming out during normal production logging - which is unexpected. 

@fxprunayre reported something similar which resulted in a good example [web/src/main/webapp/WEB-INF/classes/log4j2.xml](https://github.com/geonetwork/core-geonetwork/pull/6523#diff-eea7276aa81c6cc462ad85ec09277d485a92b1963a97144ced811fff14cdcce6) which needs to be tested and copied to the other profiles:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<Configuration status="warn" dest="out">
  <Properties>
    <Property name="log_dir">./logs</Property>
  </Properties>

  <Filters>
    <NoMarkerFilter onMatch="ACCEPT" onMismatch="NEUTRAL"/>
    <!-- filter by geonetwork module -->
    <Filters>
      <ThresholdFilter level="INFO" onMatch="NEUTRAL" onMismatch="DENY"/>
      <MarkerFilter marker="atom" onMatch="ACCEPT" onMismatch="NEUTRAL"/>
      <MarkerFilter marker="databasemigration" onMatch="ACCEPT" onMismatch="NEUTRAL"/>
      <MarkerFilter marker="encryptor" onMatch="ACCEPT" onMismatch="NEUTRAL"/>
      <MarkerFilter marker="harvester" onMatch="ACCEPT" onMismatch="NEUTRAL"/>
    </Filters>
    <Filters>
      <ThresholdFilter level="ERROR" onMatch="NEUTRAL" onMismatch="DENY"/>
      <MarkerFilter marker="geonetwork" onMatch="ACCEPT" onMismatch="NEUTRAL"/>
      <MarkerFilter marker="jeeves" onMatch="ACCEPT" onMismatch="NEUTRAL"/>
    </Filters>
  </Filters>

  <Appenders>
    <Console name="Console" target="SYSTEM_OUT">
      <PatternLayout pattern="%date{ISO8601} %-5level %marker [%logger] - %message%n"/>
    </Console>
    <RollingFile name="File">
      <filename>${sys:log_dir:-log_dir}/geonetwork.log</filename>
      <filePattern>${sys:log_dir:-log_dir}/geonetwork.log-%i.log</filePattern>
      <PatternLayout pattern="%date{ISO8601} %-5level [%logger] - %message%n"/>
      <Policies>
        <SizeBasedTriggeringPolicy size="10 MB" />
      </Policies>
      <DefaultRolloverStrategy max="3" fileIndex="min"/>
    </RollingFile>
    <Routing name="Harvester">
      <Routes pattern="$${ctx:logfile}">
        <!-- value dynamically determines the name of the log file. -->
        <Route>
          <File name="harvester-${ctx:harvester}" fileName="${sys:log_dir:-log_dir}/${ctx:logfile}">
            <PatternLayout>
              <pattern>%date{ISO8601}{${ctx:timeZone}} %-5level %marker [%logger] - %message%n</pattern>
            </PatternLayout>
          </File>
        </Route>
      </Routes>
    </Routing>
  </Appenders>

  <Loggers>
    <Logger name="org.fao.geonet" level="error" additivity="false">
      <AppenderRef ref="Console"/>
      <AppenderRef ref="File"/>
      <AppenderRef ref="Harvester"/>
    </Logger>
    <Logger name="jeeves" level="error" additivity="false">
      <AppenderRef ref="Console"/>
      <AppenderRef ref="File"/>
      <AppenderRef ref="Harvester"/>
    </Logger>

    <!--  Spring logging configuration -->
    <Logger name="org.springframework" level="error" additivity="false">
      <AppenderRef ref="Console"/>
      <AppenderRef ref="File"/>
    </Logger>
    <Logger name="org.springframework.beans" level="error"/>
    <Logger name="org.springframework.security" level="error"/>
    <Logger name="org.springframework.security.ldap" level="error"/>
    <Logger name="org.springframework.aop.framework.CglibAopProxy" level="error"/>

    <Logger name="com.k_int" level="error" additivity="false">
      <AppenderRef ref="Console"/>
      <AppenderRef ref="File"/>
    </Logger>

    <!--  JZKIT logging configuration -->
    <Logger name="org.jzkit" level="error" additivity="false">
      <AppenderRef ref="Console"/>
      <AppenderRef ref="File"/>
    </Logger>
    <Logger name="org.jzkit.a2j" level="error"/>
    <Logger name="org.jzkit.search.impl.LRUCache" level="error"/>

    <Logger name="ro.isdc.wro.http" level="error" additivity="false">
      <AppenderRef ref="Console"/>
      <AppenderRef ref="File"/>
    </Logger>

    <!-- Check domain/src/main/resources/config-spring-geonetwork.xml show_sql properties. -->
    <Logger name="org.hibernate.SQL" level="error" additivity="false">
      <AppenderRef ref="Console"/>
      <AppenderRef ref="File"/>
    </Logger>
    <Logger name="org.hibernate.type" level="error" additivity="false">
      <AppenderRef ref="Console"/>
      <AppenderRef ref="File"/>
    </Logger>
    <Logger name="org.hibernate.tool.hbm2ddl" level="error" additivity="false">
      <AppenderRef ref="Console"/>
      <AppenderRef ref="File"/>
    </Logger>
    <Logger name="org.xhtmlrenderer" level="error" additivity="false">
      <AppenderRef ref="Console"/>
      <AppenderRef ref="File"/>
    </Logger>

    <Logger name="org.apache.camel" level="error" additivity="false">
      <AppenderRef ref="Console"/>
      <AppenderRef ref="File"/>
    </Logger>

    <!-- Turn off logging except when explicitly declared above -->
    <Root level="off">
        <AppenderRef ref="File"/>
    </Root>
  </Loggers>
</Configuration>
```

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/geonetwork/core-geonetwork?pullRequest=6523) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/geonetwork/core-geonetwork?pullRequest=6523) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/geonetwork/core-geonetwork?pullRequest=6523) it.</sub>
